remotes/handlers_test.go

```diff
@@ -71,3 +77,141 @@ func TestContextCustomKeyPrefix(t *testing.T) {
+	ctx := context.Background()
+
+	out, err := SkipNonDistributableBlobs(images.HandlerFunc(func(ctx context.Context, desc ocispec.Descriptor) ([]ocispec.Descriptor, error) {
+		return []ocispec.Descriptor{
+			{MediaType: images.MediaTypeDockerSchema2Layer, Digest: "test:1"},
+			{MediaType: images.MediaTypeDockerSchema2LayerForeign, Digest: "test:2"},
+			{MediaType: images.MediaTypeDockerSchema2LayerForeignGzip, Digest: "test:3"},
+			{MediaType: ocispec.MediaTypeImageLayerNonDistributable, Digest: "test:4"},
+			{MediaType: ocispec.MediaTypeImageLayerNonDistributableGzip, Digest: "test:5"},
+			{MediaType: ocispec.MediaTypeImageLayerNonDistributableZstd, Digest: "test:6"},
+		}, nil
+	}))(ctx, ocispec.Descriptor{MediaType: images.MediaTypeDockerSchema2Manifest})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if len(out) != 1 {
+		t.Fatalf("unexpected number of descriptors returned: %d", len(out))
+	}
+	if out[0].Digest != "test:1" {
+		t.Fatalf("unexpected digest returned: %s", out[0].Digest)
+	}
+
+	dir := t.TempDir()
+	cs, err := local.NewLabeledStore(dir, newMemoryLabelStore())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	write := func(i interface{}, ref string) digest.Digest {
+		t.Helper()
+
+		data, err := json.Marshal(i)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		w, err := cs.Writer(ctx, content.WithRef(ref))
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer w.Close()
+
+		dgst := digest.SHA256.FromBytes(data)
+
+		n, err := w.Write(data)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if err := w.Commit(ctx, int64(n), dgst); err != nil {
+			t.Fatal(err)
+		}
+
+		return dgst
+	}
+
+	configDigest := write(ocispec.ImageConfig{}, "config")
+
+	manifest := ocispec.Manifest{
+		Config:    ocispec.Descriptor{Digest: configDigest, MediaType: ocispec.MediaTypeImageConfig},
+		MediaType: ocispec.MediaTypeImageManifest,
+		Layers: []ocispec.Descriptor{
+			{MediaType: images.MediaTypeDockerSchema2Layer, Digest: "test:1"},
+			{MediaType: images.MediaTypeDockerSchema2LayerForeign, Digest: "test:2"},
+			{MediaType: images.MediaTypeDockerSchema2LayerForeignGzip, Digest: "test:3"},
+			{MediaType: ocispec.MediaTypeImageLayerNonDistributable, Digest: "test:4"},
+			{MediaType: ocispec.MediaTypeImageLayerNonDistributableGzip, Digest: "test:5"},
+			{MediaType: ocispec.MediaTypeImageLayerNonDistributableZstd, Digest: "test:6"},
+		},
+	}
+
+	manifestDigest := write(manifest, "manifest")
+
+	out, err = SkipNonDistributableBlobs(images.ChildrenHandler(cs))(ctx, ocispec.Descriptor{MediaType: manifest.MediaType, Digest: manifestDigest})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if len(out) != 2 {
+		t.Fatalf("unexpected number of descriptors returned: %v", out)
+	}
+
+	if out[0].Digest != configDigest {
+		t.Fatalf("unexpected digest returned: %v", out[0])
+	}
+	if out[1].Digest != manifest.Layers[0].Digest {
+		t.Fatalf("unexpected digest returned: %v", out[1])
+	}
+}
+
+type memoryLabelStore struct {
+	l      sync.Mutex
```

Comment:
change to sync.RWMutex ?

Reply:
This is a copy/paste from the content store implementation. Only for this one test.